foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/mappings/OneToManyMapping.java

```diff
@@ -309,7 +309,12 @@ public void collectQueryParameters(Set<DatabaseField> cacheFields){
     @Override
     public Object clone() {
         OneToManyMapping clone = (OneToManyMapping)super.clone();
-        clone.setTargetForeignKeysToSourceKeys(new HashMap<>(getTargetForeignKeysToSourceKeys()));
+
+        Map<DatabaseField, DatabaseField> old2cloned = new HashMap<>();
+        clone.sourceKeyFields = cloneDatabaseFieldVector(sourceKeyFields, old2cloned);
+        clone.targetForeignKeyFields = cloneDatabaseFieldVector(targetForeignKeyFields, old2cloned);
```

Comment:
there is [cloneFields(...)](https://github.com/eclipse-ee4j/eclipselink/blob/master/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/mappings/DatabaseMapping.java#L385) method which can be used here instead of re-inventing new wheel

Reply:
sourceKeyFields and targetForeignKeyFields contain the same objects as in sourceKeysToTargetForeignKeys and targetForeignKeysToSourceKeys. cloneDatabaseFieldVector() is just a convenient way to copy new cloned objects from sourceKeyFields and targetForeignKeyFields to the sourceKeysToTargetForeignKeys and targetForeignKeysToSourceKeys as this method uses Map that binds old DatabaseField objects to the new cloned DatabaseField objects.